tests/integration-tests/tests/performance_tests/test_openfoam.py

```diff
@@ -0,0 +1,82 @@
+import json
+import logging
+
+import boto3
+import pytest
+from assertpy import assert_that
+from remote_command_executor import RemoteCommandExecutionError, RemoteCommandExecutor
+
+# timeout in seconds
+OPENFOAM_INSTALLATION_TIMEOUT = 1800
+OPENFOAM_JOB_TIMEOUT = 600
+TASK_VCPUS = 36  # vCPUs are cut in a half because multithreading is disabled
+BASELINE_CLUSTER_SIZE_ELAPSED_SECONDS = {8: 742, 16: 376, 32: 185}
+PERF_TEST_DIFFERENCE_TOLERANCE = 5
+
+
+def perf_test_difference(perf_test_result, number_of_nodes):
+    baseline_result = BASELINE_CLUSTER_SIZE_ELAPSED_SECONDS[number_of_nodes]
+    percentage_difference = 100 * (perf_test_result - baseline_result) / baseline_result
+    return percentage_difference
+
+
+def openfoam_installed(headnode):
+    cmd = '[ -d "/shared/ec2-user/SubspaceBenchmarks" ]'
+    try:
+        headnode.run_remote_command(cmd)
+        return True
+    except RemoteCommandExecutionError:
+        return False
+
+
+@pytest.mark.parametrize(
+    "number_of_nodes",
+    [[8]],
+)
+def test_openfoam(
+        vpc_stack,
+        instance,
+        os,
+        region,
+        scheduler,
+        pcluster_config_reader,
+        clusters_factory,
+        number_of_nodes,
+        test_datadir,
+        scheduler_commands_factory,
+):
+    cluster_config = pcluster_config_reader(number_of_nodes=max(number_of_nodes))
+    cluster = clusters_factory(cluster_config)
+    logging.info("Cluster Created")
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    if not openfoam_installed(remote_command_executor):
+        logging.info("Installing OpenFOAM")
+        remote_command_executor.run_remote_script(
+            str(test_datadir / "openfoam.install.sh"), timeout=OPENFOAM_INSTALLATION_TIMEOUT, hide=False
+        )
+    logging.info("OpenFOAM Installed")
+    scheduler_commands = scheduler_commands_factory(remote_command_executor)
+    performance_degradation = {}
+    subspace_benchmarks_dir = "/shared/ec2-user/SubspaceBenchmarks"
+    for node in number_of_nodes:
+        result = remote_command_executor.run_remote_command(
+            f'bash openfoam.slurm.sh "{subspace_benchmarks_dir}" "{node}"',
+            additional_files=[str(test_datadir / "openfoam.slurm.sh")],
+        )
+        logging.info(f"Submitting OpenFOAM job with {node} nodes")
+        job_id = scheduler_commands.assert_job_submitted(result.stdout)
+        scheduler_commands.wait_job_completed(job_id, timeout=OPENFOAM_JOB_TIMEOUT)
+        scheduler_commands.assert_job_succeeded(job_id)
+        perf_test_result = remote_command_executor.run_remote_script(
```

Comment:
The result file collects one measurement per execution. This means that when you run it for 8 nodes you will have one line in the csv file, but when you run 16 nodes you will have two lines. The script must take this into account otherwise it will not return the correct value for the elapsed time.